modules/aws-eks-iam-role-group/main.tf

```
@@ -27,6 +27,20 @@ data "aws_iam_policy_document" "assume_role_policy" {
     }
     actions = ["sts:AssumeRole"]
   }
+
+  dynamic "statement" {
+    for_each = var.external_arn_role
```

I'd name it `extra_role_principals` or `additional_role_principals` - IMO it says more than `external_arn_role` (which implies that one can create external role with specific ARN - I didn't get it at first sight)

```suggestion
    for_each = var.additional_role_principals
```